### ISSUE TITLE
0xGoodess - user can queue an withdrawal request bigger than their current withdrawable to extract MEV from the system

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-04-jojo-judging#25]
0xGoodess

high

# user can queue an withdrawal request bigger than their current withdrawable to extract MEV from the system

## Summary
user can queue an withdrawal request bigger than their withdrawable to extract MEV from the system

## Vulnerability Detail
requestWithdraw does not checks the request amount, whether it is within the current withdrawable amount of the user. User can always queue up an withdrawal first before actually depositing / start trading. The enforced interval becomes in-effective.

The consequence can be quite severe. Given a user who would like to game this; he would first queue 100 USDC withdrawal, and deposit 100 USDC into the system only after the withdrawalLock passes. Then he could open, say a long position. If the next oracle update is a price reduction, he could frontrun with immediate withdrawal; if the next oracle update is a price appreciation, he could simply keep the position.  The trader can systematically risk nothing ,to get unlimited upside.

```solidity
function requestWithdraw(
        Types.State storage state,
        uint256 primaryAmount,
        uint256 secondaryAmount
    ) external {
        state.pendingPrimaryWithdraw[msg.sender] = primaryAmount;
        state.pendingSecondaryWithdraw[msg.sender] = secondaryAmount;
        state.withdrawExecutionTimestamp[msg.sender] =
            block.timestamp +
            state.withdrawTimeLock;
```

## Impact
The enforced pending interval becomes in-effective.

## Code Snippet
https://github.com/sherlock-audit/2023-04-jojo/blob/main/smart-contract-EVM/contracts/lib/Funding.sol#L84-L100

## Tool used

Manual Review

## Recommendation

1. enforce a cap on the requestedAmount, for example the current primaryCredit + netPositionValue etc.
2. Once a position is opened, the requestAmount needs to be updated to reflect the latest withdrawable 

Duplicate of #429
